### PR TITLE
mlx: Improve gemma4 performance with fused operations

### PR DESCRIPTION
**Pull request by dhiltgen**
`go run cmd/bench/bench.go -model gemma4:XXX-nvfp4 -prompt-tokens 2048 -max-tokens 128 -epochs 5 -warmup 1`

```
  ┌──────┬─────────┬──────────┬────────────────┬────────┐
  │ Size │ Metric  │ Baseline │ New (compiled) │   Δ    │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ E2B  │ prefill │   18,777 │         21,901 │ +16.6% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ E2B  │ gen     │    153.8 │          176.9 │ +15.0% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ E4B  │ prefill │    6,980 │          8,086 │ +15.8% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ E4B  │ gen     │     99.1 │          110.6 │ +11.6% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ 26B  │ prefill │    3,957 │          4,372 │ +10.5% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ 26B  │ gen     │    101.3 │          107.5 │  +6.1% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ 31B  │ prefill │      531 │            593 │ +11.7% │
  ├──────┼─────────┼──────────┼────────────────┼────────┤
  │ 31B  │ gen     │     21.4 │           22.4 │  +4.8% │
  └──────┴─────────┴──────────┴────────────────┴────────┘
```

### CODE REVIEW

**Review by pdevine — APPROVED** [review]

File: x/models/gemma4/gemma4.go [hunk 1]
```diff
@@ -1114,9 +1114,8 @@ func (m *Model) Unembed(x *mlx.Array) *mlx.Array {
 	logits := m.LMHead.Forward(x)
 
 	if m.FinalLogitSoftcapping > 0 {
-		logits = mlx.MulScalar(logits, m.SoftcapInv)
```
Comment (jessegross) [inline x/models/gemma4/gemma4.go:1117]: `m.SoftcapInv` is now unused, I think.

**Review by jessegross — APPROVED** [review]
